flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/JobMaster.java

```diff
@@ -333,10 +340,15 @@ public void onUnknownDeploymentsOf(
     }
 
     private SchedulerNG createScheduler(
+            Configuration configuration,
             ExecutionDeploymentTracker executionDeploymentTracker,
             JobManagerJobMetricGroup jobManagerJobMetricGroup,
             JobStatusListener jobStatusListener)
             throws Exception {
+
+        List<FailureListener> failureListeners =
+                FailureListenerUtils.getFailureListerners(configuration, jobManagerJobMetricGroup);
```

Comment:
I think it would be better to initialize the failure listeners in the JobMaster constructor, and pass the set into this method. In the JobMaster constructor, you have access to all the required fields (jobid, jobname, config etc.)

Reply:
Resolved